### PR TITLE
Adding iOS Build and Test to CI

### PR DESCRIPTION
**Pull request by fluffyemily**

### CODE REVIEW

**Comment by ncalexan** [comment]
It looks like the build directories with all of the Rust targets are too big for Travis CI's build caching.  Should we `cargo clean`, so that we keep VCS checkouts but drop object files?  Can we split targets across jobs?

File: .travis.yml [hunk 1]
```diff
@@ -24,6 +34,14 @@ script:
     for manifest in $(find . -type f -name Cargo.toml); do
       cargo test --manifest-path $manifest --verbose --no-default-features --features sqlcipher
     done
+  - |
+    if [[ "$TRAVIS_RUST_VERSION" = "1.25.0" ]]; then
+      cd ffi &&
```
Comment (ncalexan) [inline .travis.yml]: Consider `pushd` and `popd`.

File: .travis.yml [hunk 2]
```diff
@@ -13,6 +13,16 @@ matrix:
   allow_failures:
     - rust: nightly
   fast_finish: true
+install:
+  - |
+    if [[ "$TRAVIS_RUST_VERSION" = "1.25.0" ]]; then
+      rustup target add aarch64-apple-ios armv7-apple-ios armv7s-apple-ios x86_64-apple-ios i386-apple-ios &&
```
Comment (ncalexan) [inline .travis.yml]: That's a lot of targets.  Do we actually run tests on all of these targets?

File: .travis.yml [hunk 3]
```diff
@@ -13,6 +13,16 @@ matrix:
...
+      # Cargo-lipo needs to be Beta version available on master and not crates version to handle building in subdirectories
+      cargo install --git https://github.com/TimNN/cargo-lipo &&
```
Comment (ncalexan) [inline .travis.yml]: Is it possible to pin this version to a SHA?  We shouldn't be fetching something random off the Web, basically ever.

File: .travis.yml [hunk 4]
```diff
@@ -13,6 +13,16 @@ matrix:
...
+      # If we don't list the devices available then when we come to pick on during the test run travis can't find our device.
```
Comment (ncalexan) [inline .travis.yml]: nit: s/pick on/pick one/.
nit: full sentence with punctuation, here and above.

File: .travis.yml [hunk 5]
```diff
@@ -24,6 +34,14 @@ script:
     for manifest in $(find . -type f -name Cargo.toml); do
       cargo test --manifest-path $manifest --verbose --no-default-features --features sqlcipher
     done
+  - |
+    if [[ "$TRAVIS_RUST_VERSION" = "1.25.0" ]]; then
```
Comment (ncalexan) [inline .travis.yml]: Why are we only doing this for one Rust version?

File: .travis.yml (hunk 2 above)
Comment (fluffyemily) [reply .travis.yml]: We build a universal library that contains all of the targets so we do need to add all of them.

**Pull request merged**